modules/schema/src/main/java/org/apache/ignite/internal/schema/marshaller/MarshallerUtil.java

```diff
@@ -45,32 +46,31 @@
     public static int getValueSize(Object val, NativeType type) throws InvalidTypeException {
         switch (type.spec()) {
             case BYTES:
-                return ((byte[]) val).length;
-            
+                // Return zero for pojo as they are not serialized yet.
+                return (val instanceof byte[]) ? ((byte[]) val).length : 0;
```

Comment:
Is it worth setting some small value to guess object size? With zero we'll underestimate it in 100% cases, but with 100 bytes (some constant) we have a chance to guess.

Reply:
I'm not sure we can guess anything here, it depends on serialization. Also, it is easy to over-estimate if there are many fields with pojos. Maybe we can disable row size calculation if there is non-null field with user object and use predefined initial buffer size.

As an alternative, we can try to serialize object to a 'blackhole' using configured serializer just to count result byte[] size (without any allocation). This requires some benchmarks.